src/utils/HashlinkEncoder.ts

```diff
@@ -37,17 +40,16 @@ export class HashlinkEncoder {
    */
   public static encode(
     buffer: Buffer | Uint8Array,
-    hashAlgorithm: 'sha2-256',
-    baseEncoding: BaseName = 'base58btc',
-    metadata?: Metadata
+    metadata?: Metadata,
+    hashAlgorithm: 'sha2-256' = 'sha2-256',
+    baseEncoding: BaseName = 'base58btc'
   ) {
-    const checksum = this.encodeMultihashEncoder(buffer, hashAlgorithm, baseEncoding)
-    const mbMetadata = metadata ? this.encodeMetadata(metadata, baseEncoding) : null
+    const checksum = this.encodeMultiHashEncoder(buffer, hashAlgorithm, baseEncoding)
```

Comment:
This name seems weird. We don't want to encode the multiHashEncoder right?

```suggestion
    const checksum = this.encodeMultiHash(buffer, hashAlgorithm, baseEncoding)
```